lib/protobuf/field/varint_field.rb

```diff
@@ -37,11 +37,17 @@ def self.encode(value)
       #
 
       def acceptable?(val)
-        (val > self.class.min || val < self.class.max)
+        int_val = coerce!(val)
+        int_val >= self.class.min && int_val <= self.class.max
       rescue
         false
       end
 
+      def coerce!(val)
+        return val.to_i if val.is_a?(Numeric)
```

Comment:
Is this required? (the .to_i and is_a?(Numeric) check)

we have been looking at standardizing serialization with this branch
https://github.com/localshred/protobuf/compare/bdewitt/fix_coercion